### PR TITLE
[190] Adding better checks when multioutputTransactionScenario

### PR DESCRIPTION
**Pull request by paweljakubas**
<p align="right">#190</p>

# Overview

- [x] I have added two element list type to better check in multioutputTransactionScenario function

# Comments

### CODE REVIEW

File: test/integration/Test/Integration/Framework/Auxiliary.hs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import qualified Data.List as L
+
+data FixedSizedList (n :: Nat) a = UnsafeMkList { getList :: [a] }
+    deriving Show
+
+type TwoElementList a = FixedSizedList 2 a
+
+mkFixedSizedList :: forall n a. KnownNat n => [a] -> Maybe (FixedSizedList n a)
+mkFixedSizedList v | L.length v == l = Just (UnsafeMkList v)
+                   | otherwise       = Nothing
```
Comment (KtorZ) [inline test/integration/Test/Integration/Framework/Auxiliary.hs]: Minor things but: no more variable-length indentation please :pray:

File: test/integration/Test/Integration/Scenario/Transactions.hs [hunk 2]
```diff
@@ -344,35 +345,42 @@ spec = do
     multioutputTransactionScenario
         :: String
         -> [Word64]
-        -> [Word64]
+        -> Maybe (TwoElementList Word64)
```
Comment (KtorZ) [inline test/integration/Test/Integration/Scenario/Transactions.hs]: This shouldn't be a `Maybe`; this is made fairly obvious by the way the function handles that with a `case` right away, failing hard in the case it receives a `Nothing`. Generally-speaking, make the input domain no more than what you need. 
This is up to the caller to produce a value of the right type.

**Review by KtorZ — CHANGES_REQUESTED** [review]
I think we are a bit over-doing it here. Since this is just a local function and a fairly limited scope, it sounds a bit too much albeit being indeed more type-safe. 
Though, using just a 2-tuple or a record or a dedicated type would work the same.  

In any case, I think we could simply get away with a clear invariant, before executing the scenario, verifying a bunch of precondition on the inputs. If not met, it fails hard with a clear and helpful error message. This code runs only in our testsuite and we should strive to keep it rather simple; still, making sure we can debug it easily is of the utmost importance.

**Comment by paweljakubas** [comment]
thanks for the remarks! I reconsidered the issue and finally opted for 2-tuple (although still think FixedSizedList has the potential to be used in the future but needs more helper functions to be added)

File: test/integration/Test/Integration/Scenario/Transactions.hs (hunk 2 above)
Comment (paweljakubas) [reply test/integration/Test/Integration/Scenario/Transactions.hs]: yes, the caller should take responsibility here

**Review by KtorZ — APPROVED** [review]
:tada:

**Pull request merged**